File: projects/NativeQueueSample/source/NativeQueueSample.Core/Class1.cpp
//-----------------------------------------------------------------------
// <copyright file="Class1.cpp" company="Brian Rogers">
// Copyright (c) Brian Rogers. All rights reserved.
// </copyright>
//-----------------------------------------------------------------------

#include "Class1.h"

using namespace concurrency;
using namespace std;

namespace NativeQueueSample
{
    Class1::Class1(wstring const & name)
        : name_(name)
    {
    }

    Class1::~Class1()
    {
    }

    task<wstring> Class1::DoAsync() const
    {
        return task_from_result(name_);
    }
}

File: projects/TaskSample/TaskSample.Core/Class1.cs
// <copyright file="Class1.cs" company="Brian Rogers">
// Copyright (c) Brian Rogers. All rights reserved.
// </copyright>

namespace TaskSample
{
    public class Class1
    {
        private readonly string name;

        public Class1(string name)
        {
            this.name = name;
        }

        public override string ToString() => this.name;
    }
}
